openquake/hazardlib/calc/filters.py

```diff
@@ -122,6 +122,125 @@ def filter_sites_by_distance_to_rupture(rupture, integration_distance, sites):
+        dist = rupture.surface.get_ry0_distance(mesh)
+    elif param == 'rjb':
+        dist = rupture.surface.get_joyner_boore_distance(mesh)
+    elif param == 'rhypo':
+        dist = rupture.hypocenter.distance_to_mesh(mesh)
+    elif param == 'repi':
+        dist = rupture.hypocenter.distance_to_mesh(mesh, with_depths=False)
+    elif param == 'rcdpp':
+        dist = rupture.get_cdppvalue(mesh)
+    elif param == 'azimuth':
+        dist = rupture.surface.get_azimuth(mesh)
+    else:
+        raise ValueError('Unknown distance measure %r' % param)
+    return dist
+
+
+class FarAwayRupture(Exception):
+    """Raised if the rupture is outside the maximum distance for all sites"""
+
+
+class IntegrationDistance(collections.Mapping):
+    """
+    Pickleable object wrapping a dictionary of integration distances per
+    tectonic region type. The integration distances can be scalars or
+    list of pairs (magnitude, distance). Here is an example using 'default'
+    as tectonic region type sot that the same values will be used for all
+    tectonic region types:
+
+    >>> maxdist = IntegrationDistance({'default': [
+    ...          (1, 10), (2, 20), (3, 30), (4, 40), (5, 100), (6, 200),
+    ...          (7, 400), (8, 800)]})
+    >>> maxdist('Some TRT', mag=5.5)
+    array(150.0)
+
+    It has also a method `.get_closest(sites, rupture)` returning the closest
+    sites to the rupture and their distances. The integration distance can be
+    missing if the sites have been already filtered (empty dictionary): in
+    that case the method returns all the sites and all the distances.
+    """
+    def __init__(self, dic):
+        self.dic = dic or {}  # TRT -> float or list of pairs
+        self.magdist = {}  # TRT -> (magnitudes, distances)
+        for trt, value in self.dic.items():
+            if isinstance(value, list):  # assume a list of pairs (mag, dist)
+                value.sort()  # make sure the list is sorted by magnitude
+                self.magdist[trt] = zip(*value)
+            else:
+                self.dic[trt] = float(value)
+
+    def __call__(self, trt, mag=None):
+        value = getdefault(self.dic, trt)
+        if isinstance(value, float):  # scalar maximum distance
+            return value
+        elif mag is None:  # get the maximum magnitude distance
+            return value[-1][1]
+        elif not hasattr(self, 'interp'):
+            self.interp = {}  # function cache
+        try:
+            md = self.interp[trt]  # retrieve from the cache
+        except KeyError:  # fill the cache
+            magdist = getdefault(self.magdist, trt)
+            md = self.interp[trt] = interp1d(
+                *magdist, bounds_error=False, fill_value='extrapolate')
+        return md(mag)
+
+    def get_closest(self, sites, rupture, distance_type='rrup'):
+        """
+        :param sites: a (Filtered)SiteColletion
+        :param rupture: a rupture
+        :param distance_type: default 'rjb'
```

Comment:
Please fix the default in the docstring or in the signature, to make them consistent